File: VB/IPhoneCheckEditSample/IPhoneCheckEdit/IPhoneCheckEditViewInfo.vb
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo

Namespace DevExpress.WinForms.Samples.IPhoneCheckEdit
	Public Class IPhoneCheckEditViewInfo
		Inherits CheckEditViewInfo
		Private infoText_Renamed As String
		Private backgroundImageIndex_Renamed As Integer
		Private buttonImageIndex_Renamed As Integer
		Private backgroundRect_Renamed As Rectangle
		Private buttonRect_Renamed As Rectangle
		Private infoTextLocation_Renamed As Point

		Public ReadOnly Property InfoText() As String
			Get
				Return infoText_Renamed
			End Get
		End Property
		Public ReadOnly Property InfoTextLocation() As Point
			Get
				Return infoTextLocation_Renamed
			End Get
		End Property
		Public Shadows ReadOnly Property Item() As RepositoryItemIPhoneCheckEdit
			Get
				Return TryCast(MyBase.Item, RepositoryItemIPhoneCheckEdit)
			End Get
		End Property
		Public ReadOnly Property BackgroundRect() As Rectangle
			Get
				Return backgroundRect_Renamed
			End Get
		End Property
		Public ReadOnly Property ButtonRect() As Rectangle
			Get
				Return buttonRect_Renamed
			End Get
		End Property
		Public ReadOnly Property BackgroundImageIndex() As Integer
			Get
				Return backgroundImageIndex_Renamed
			End Get
		End Property
		Public ReadOnly Property ButtonImageIndex() As Integer
			Get
				Return buttonImageIndex_Renamed
			End Get
		End Property

		Public Sub New(ByVal item As RepositoryItem)
			MyBase.New(item)
		End Sub

		Public Overrides Property AllowDrawFocusRect() As Boolean
			Get
				Return False
			End Get
			Set(ByVal value As Boolean)
				MyBase.AllowDrawFocusRect = False
			End Set
		End Property
		Public Overrides Overloads Sub Offset(ByVal x As Integer, ByVal y As Integer)
			MyBase.Offset(x, y)
			OffsetContentLocation(x, y)
		End Sub
		Public Overrides Sub UpdatePaintAppearance()
			If CheckState = System.Windows.Forms.CheckState.Checked Then
				PaintAppearance.Assign(Item.AppearanceChecked)
				Return
			End If
			MyBase.UpdatePaintAppearance()
		End Sub
		Protected Overrides Sub Assign(ByVal info As BaseControlViewInfo)
			MyBase.Assign(info)
			Dim view As IPhoneCheckEditViewInfo = (TryCast(info, IPhoneCheckEditViewInfo))
			If view Is Nothing Then
				Return
			End If
			Me.infoText_Renamed = view.infoText
			Me.buttonImageIndex_Renamed = view.backgroundImageIndex
			Me.buttonImageIndex_Renamed = view.buttonImageIndex
			Me.backgroundRect_Renamed = view.backgroundRect
			Me.buttonRect_Renamed = view.buttonRect
			Me.infoTextLocation_Renamed = view.infoTextLocation
		End Sub
		Protected Overrides Sub CalcContentRect(ByVal bounds As Rectangle)
			MyBase.CalcContentRect(bounds)
			CalcBackgroundRectSize()
			CalcButtonRectSize()
			UpdateButtonLocationByState()
			UpdateInfoTextLocationByState()
		End Sub
		Protected Overrides Sub OnEditValueChanged()
			MyBase.OnEditValueChanged()
			UpdateBackgroundSkinImageIndexByState()
			UpdateButtonSkinImageIndexByState()
			UpdateButtonLocationByState()
			UpdateInfoTextByState()
			UpdateInfoTextLocationByState()
		End Sub

		Private Sub CalcBackgroundRectSize()
			Me.backgroundRect_Renamed.Size = ContentRect.Size
		End Sub
		Private Sub CalcButtonRectSize()
			Me.buttonRect_Renamed.Size = New Size(ContentRect.Width \ 2, ContentRect.Height)
		End Sub
		Private Sub OffsetContentLocation(ByVal x As Integer, ByVal y As Integer)
			Me.backgroundRect_Renamed.Offset(x, y)
			Me.buttonRect_Renamed.Offset(x, y)
			Me.infoTextLocation_Renamed.Offset(x, y)
		End Sub
		Private Sub UpdateBackgroundSkinImageIndexByState()
			Me.backgroundImageIndex_Renamed = If((CheckState = CheckState.Checked), Item.BackgroundSkinCheckedImageIndex, Item.BackgroundSkinUncheckedImageIndex)
		End Sub
		Private Sub UpdateButtonSkinImageIndexByState()
			Me.buttonImageIndex_Renamed = Item.ButtonSkinImageIndex
		End Sub
		Private Sub UpdateButtonLocationByState()
			Me.buttonRect_Renamed.Y = ContentRect.Top
			Me.buttonRect_Renamed.X = If((CheckState = CheckState.Checked), (ContentRect.Right - Me.buttonRect_Renamed.Width), ContentRect.Left)
		End Sub
		Private Sub UpdateInfoTextByState()
			Me.infoText_Renamed = If((CheckState = CheckState.Checked), Item.InfoTextChecked, Item.InfoTextUnchecked)
		End Sub
		Private Sub UpdateInfoTextLocationByState()
            Me.infoTextLocation_Renamed.Y = ContentRect.Top + CType((ContentRect.Height / 2), Integer)
            If CheckState = CheckState.Checked Then
                Me.infoTextLocation_Renamed.X = ContentRect.Left + CType((Me.buttonRect_Renamed.Left / 2), Integer)
            Else
                Me.infoTextLocation_Renamed.X = ContentRect.Right - CType(((ContentRect.Right - Me.buttonRect_Renamed.Right) / 2), Integer)
            End If
        End Sub
	End Class
End Namespace

File: CS/IPhoneCheckEditSample/IPhoneCheckEdit/IPhoneCheckEditViewInfo.cs
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;

namespace DevExpress.WinForms.Samples.IPhoneCheckEdit {
    public class IPhoneCheckEditViewInfo : CheckEditViewInfo {
        string infoText;
        int backgroundImageIndex;
        int buttonImageIndex;
        Rectangle backgroundRect;
        Rectangle buttonRect;
        Point infoTextLocation;

        public string InfoText { get { return infoText; } }
        public Point InfoTextLocation { get { return infoTextLocation; } }
        public new RepositoryItemIPhoneCheckEdit Item { get { return base.Item as RepositoryItemIPhoneCheckEdit; } }
        public Rectangle BackgroundRect { get { return backgroundRect; } }
        public Rectangle ButtonRect { get { return buttonRect; } }
        public int BackgroundImageIndex { get { return backgroundImageIndex; } }
        public int ButtonImageIndex { get { return buttonImageIndex; } }

        public IPhoneCheckEditViewInfo(RepositoryItem item) : base(item) {
        }

        public override bool AllowDrawFocusRect {
            get { return false; }
            set { base.AllowDrawFocusRect = false; }
        }
        public override void Offset(int x, int y) {
            base.Offset(x, y);
            OffsetContentLocation(x, y);
        }
        public override void UpdatePaintAppearance() {
            if(CheckState == System.Windows.Forms.CheckState.Checked) {
                PaintAppearance.Assign(Item.AppearanceChecked);
                return;
            }
            base.UpdatePaintAppearance();
        }
        protected override void Assign(BaseControlViewInfo info) {
            base.Assign(info);
            IPhoneCheckEditViewInfo view = (info as IPhoneCheckEditViewInfo);
            if(view == null) return;
            this.infoText = view.infoText;
            this.buttonImageIndex = view.backgroundImageIndex;
            this.buttonImageIndex = view.buttonImageIndex;
            this.backgroundRect = view.backgroundRect;
            this.buttonRect = view.buttonRect;
            this.infoTextLocation = view.infoTextLocation;
        }
        protected override void CalcContentRect(Rectangle bounds) {
            base.CalcContentRect(bounds);
            CalcBackgroundRectSize();
            CalcButtonRectSize();
            UpdateButtonLocationByState();
            UpdateInfoTextLocationByState();
        }
        protected override void OnEditValueChanged() {
            base.OnEditValueChanged();
            UpdateBackgroundSkinImageIndexByState();
            UpdateButtonSkinImageIndexByState();
            UpdateButtonLocationByState();
            UpdateInfoTextByState();
            UpdateInfoTextLocationByState();
        }

        void CalcBackgroundRectSize() {
            this.backgroundRect.Size = ContentRect.Size;
        }
        void CalcButtonRectSize() {
            this.buttonRect.Size = new Size(ContentRect.Width / 2, ContentRect.Height);
        }
        void OffsetContentLocation(int x, int y) {
            this.backgroundRect.Offset(x, y);
            this.buttonRect.Offset(x, y);
            this.infoTextLocation.Offset(x, y);
        }
        void UpdateBackgroundSkinImageIndexByState() {
            this.backgroundImageIndex = (CheckState == CheckState.Checked) ? Item.BackgroundSkinCheckedImageIndex : Item.BackgroundSkinUncheckedImageIndex;
        }
        void UpdateButtonSkinImageIndexByState() {
            this.buttonImageIndex = Item.ButtonSkinImageIndex;
        }
        void UpdateButtonLocationByState() {
            this.buttonRect.Y = ContentRect.Top;
            this.buttonRect.X = (CheckState == CheckState.Checked) ? (ContentRect.Right - this.buttonRect.Width) : ContentRect.Left;
        }
        void UpdateInfoTextByState() {
            this.infoText = (CheckState == CheckState.Checked) ? Item.InfoTextChecked : Item.InfoTextUnchecked;
        }
        void UpdateInfoTextLocationByState() {
            this.infoTextLocation.Y = ContentRect.Top + (ContentRect.Height / 2);
            this.infoTextLocation.X = (CheckState == CheckState.Checked) ? (ContentRect.Left + (this.buttonRect.Left / 2)) : (ContentRect.Right - ((ContentRect.Right - this.buttonRect.Right) / 2));
        }
    }
}
